probe-rs/src/architecture/arm/sequences/cc13xx_cc26xx.rs

```diff
@@ -0,0 +1,340 @@
+//! Sequences for cc13xx_cc26xx devices
+
+use std::sync::Arc;
+
+use super::{ArmDebugSequence, ArmDebugSequenceError};
+use crate::architecture::arm::communication_interface::DapProbe;
+use crate::architecture::arm::ArmError;
+
+/// Marker struct indicating initialization sequencing for cc13xx_cc26xx family parts.
+pub struct CC13xxCC26xx {}
+
+// IR register values, see https://www.ti.com/lit/ug/swcu185f/swcu185f.pdf table 6-7
+const IR_ROUTER: u64 = 0x02;
+const IR_CONNECT: u64 = 0x07;
+const IR_BYPASS: u64 = 0x3F;
+const IR_LEN_IN_BITS: u8 = 6;
+
+#[derive(PartialEq)]
+enum JtagState {
+    RunTestIdle = 0x1,
+    SelectDRScan = 0x2,
+    ShiftDR = 0x03,
+    ShiftIR = 0x04,
+}
+
+// Set the bottom n bits of a u64 to 1
+// This is lifted directly from:
+// https://users.rust-lang.org/t/how-to-make-an-integer-with-n-bits-set-without-overflow/63078/6
+fn set_n_bits(x: u32) -> u64 {
+    u64::checked_shl(1, x).unwrap_or(0).wrapping_sub(1)
+}
+
+impl CC13xxCC26xx {
+    /// Create the sequencer for the cc13xx_cc26xx family of parts.
+    pub fn create() -> Arc<Self> {
+        Arc::new(Self {})
+    }
+
+    /// This function implements a Zero Bit Scan(ZBS)
+    ///
+    /// The ZBS defined in section 6.2.2.1 of this document:
+    /// https://www.ti.com/lit/ug/swcu185f/swcu185f.pdf
+    ///
+    /// This function assumes that the JTAG state machine is in the Run-Test/Idle state
+    ///
+    /// * `interface` - Reference to interface to interact with CmsisDap
+    fn zero_bit_scan(&self, interface: &mut dyn DapProbe) -> Result<(), ArmError> {
+        // Enter DRSELECT state
+        interface.jtag_sequence(1, true, 0x01)?;
```

Comment:
This assumes `cmsisdap`, which is fine, I need to communicate with the debug probe, but it does close the door for using `probe-rs` + cc13xx_cc26xx + non `cmsisdap` probes.

Is there a clean way to abstract this?

Reply:
We have an error that we throw when a specific probe is not supported. I think it's a valid thing. ST-Link simply has not the same capabilities than J-Link and CMSIS-DAP.
The same should work for J-Link as well :)

Reply:
I think this one is solved now, let me know what you think 😃

Reply:
Thanks! Looks good :)